Invalid pointer reference in back_emplace?

**Describe the bug**

Compile problem when using `back_emplace` in a `std::transform()` call:

```
/root/plugin/build/_deps/apex-src/include/apex/core/iterator.hpp:96:20: error: member reference type 'apex::back_emplacer<std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >::container_type *' (aka 'std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > *') is a pointer; did you mean to use '->'? [clang-diagnostic-error]
    this->container.emplace_back(std::forward<T>(t));
```

**To Reproduce**
See discussion from https://github.com/netlify/netlify-ats-plugin/pull/346
(eliding code due to public repo)

--- a/include/apex/core/iterator.hpp
+++ b/include/apex/core/iterator.hpp
@@ -89,11 +89,10 @@ struct back_emplacer {
   { }
 
   template <class T>
-  back_emplacer& operator = (T&& t)
-    noexcept(noexcept(this->container->emplace_back(std::forward<T>(t))))
-    requires requires (container_type* container) { container->emplace_back(std::forward<T>(t)); }
+  requires requires (container_type* container, T&& x) { container->emplace_back(static_cast<T&&>(x)); }
+  back_emplacer& operator = (T&& x) noexcept(noexcept(this->container->emplace_back(static_cast<T&&>(x))))
   {
-    this->container.emplace_back(std::forward<T>(t));
+    this->container->emplace_back(static_cast<T&&>(x));
     return *this;
   }
 

--- a/tests/core/iterator.cxx
+++ b/tests/core/iterator.cxx
@@ -1,0 +1,14 @@
+#include <apex/core/iterator.hpp>
+#include <algorithm>
+#include <vector>
+
+TEST_CASE("back_emplacer") {
+  std::vector<int> x { 1, 2, 3, 4 };
+  std::vector<int> y;
+  std::transform(x.begin(), x.end(), apex::back_emplacer(y), [] (auto v) { return v + 1; });
+  REQUIRE(y.size() == x.size());
+  REQUIRE(x[0] == (y[0] - 1));
+  REQUIRE(x[1] == (y[1] - 1));
+  REQUIRE(x[2] == (y[2] - 1));
+  REQUIRE(x[3] == (y[3] - 1));
+}

